index.js

```diff
@@ -195,15 +195,44 @@ DecryptionStream.prototype._transform = function(chunk, enc, cb) {
 };
 
 DecryptionStream.prototype._flush = function(cb) {
-  var data = Buffer.concat(this._cipherTextChunks);// this could be rewritten to avoid doing this
-  var mac = data.slice(-16);
+  var mac = pullOutMac(this._cipherTextChunks);
+  if (!mac) {
+    return this.emit('error', new Error('insufficient data'));
```

Comment:
We should either be more specific with this error message, like: `Insufficient data for decryption: no mac present.` or super vague: `Decryption failed: bad cipher text.`